pkg/client/connector/traffic_manager.go

```diff
@@ -499,3 +511,102 @@ func (tm *trafficManager) uninstall(c context.Context, ur *rpc.UninstallRequest)
 	}
 	return result, nil
 }
+
+var svcCIDRrx = regexp.MustCompile(`range of valid IPs is (.*)$`)
+
+// getClusterCIDRs finds the service CIDR and the pod CIDRs of all nodes in the cluster
+func (tm *trafficManager) getOutboundInfo(c context.Context, mgrPort int32) (*daemon.OutboundInfo, error) {
+	// Get all nodes
+	var nodes []kates.Node
+	var cidr *net.IPNet
+
+	dnsServices := []metav1.ObjectMeta{
+		{
+			Name:      "kube-dns",
+			Namespace: "kube-system",
+		},
+		{
+			Name:      "dns-default",
+			Namespace: "openshift-dns",
+		},
+	}
+	var kubeDNS net.IP
+	for _, dnsService := range dnsServices {
+		svc := kates.Service{TypeMeta: metav1.TypeMeta{Kind: "Service"}, ObjectMeta: dnsService}
+		if err := tm.client.Get(c, &svc, &svc); err == nil {
+			kubeDNS = iputil.Parse(svc.Spec.ClusterIP)
+			break
+		}
+	}
```

Comment:
Part of me thinks we should just grab the IP from the `nameserver` line in `/etc/resolv.conf`.

Reply:
Yes, let's do that when we move the subnet discovery to the traffic-manager service.